x/mongo/driver/ocsp/ocsp.go

```diff
@@ -52,8 +52,28 @@ func newOCSPError(wrapped error) error {
 	return &Error{wrapped: wrapped}
 }
 
+// Response contains a subset of the details needed from an OCSP response after the original response has been
+// validated.
+type Response struct {
+	Status     int
+	NextUpdate time.Time
+}
+
+func newResponse(res *ocsp.Response) *Response {
+	return &Response{
+		Status:     res.Status,
+		NextUpdate: res.NextUpdate,
+	}
+}
+
 // Verify performs OCSP verification for the provided ConnectionState instance.
-func Verify(ctx context.Context, connState tls.ConnectionState) error {
+func Verify(ctx context.Context, connState tls.ConnectionState, opts *VerifyOptions) error {
+	if opts.Cache == nil {
+		// There should always be an OCSP cache. Even if the user has specified the URI option to disable communication
+		// with OCSP responders, the driver will cache any stapled responses. Requiring that the cache is non-nil
+		// allows us to confirm that the cache is correctly being passed down from a higher level.
+		return newOCSPError(errors.New("no OCSP cache provided"))
```

Comment:
Our panic handling has caused some issues in the past when panics happen in background routines, so I'd prefer to keep this as an error right now to avoid issues if the panic happens in the heartbeat routine.